shark-graph/src/main/java/shark/internal/Tuples.kt

```diff
@@ -0,0 +1,69 @@
+package shark.internal
+
+/**
+ * Replacement to a traditional Pair<Long, Object> that doesn't box long.
+ * Class is purposefully not made as a `data class` to decrease memory footprint of the object.
```

Comment:
Based on allocation view `LongPair` takes 24 bytes, while regular `Pair` is 16. Makes sense, since `LongPair` contains `long` in it (8 bytes), while `Pair` references another `Long` object.